lib/utils/new-module.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const EMBER_NAMESPACES = ['inject.controller', 'inject.service'];
+
+function buildMessage(obj) {
+  // Both the Controller module and Service module each make available the
+  // "inject" namespace. In order to make it more readable, so it's more
+  // explicit at first glance from which module the "inject" namespace
+  // belongs to or is being imported from, let's check against this so we
+  // can properly populate the import specifier to report the ESLint error.
+  // Ex: import { inject as service } from '@ember/service';
+  const isNamespace = EMBER_NAMESPACES.indexOf(`${obj.parent}.${obj.key}`) !== -1;
+
+  const isNamedExport = obj.match.export !== 'default';
+
+  let importSpecifier;
+  let message;
+
+  if (isNamedExport && !isNamespace) {
+    importSpecifier =
+      obj.match.localName && obj.match.localName !== obj.match.export
+        ? `{ ${obj.match.export} as ${obj.match.localName} }`
+        : `{ ${obj.match.export} }`;
+  } else if (isNamedExport && isNamespace) {
+    importSpecifier = `{ ${obj.parent} as ${obj.key} }`;
+  } else {
+    importSpecifier = obj.match.localName || obj.customKey || obj.key;
+  }
+
+  const replacement = `import ${importSpecifier} from '${obj.match.module}';`;
+
+  if (obj.type === 'Property') {
+    const global = obj.match.global.match(/^DS/) ? 'DS' : 'Ember';
+    message = `Use \`${replacement}\` instead of using ${global} destructuring`;
+  } else {
+    message = `Use \`${replacement}\` instead of using ${obj.fullName}`;
+  }
+
+  return message;
+}
+
+/**
+ * This function returns a fix function. The fix function can update the old
+ * imports to the new ones.
+ * @param {Object} node - the AST node related to the import problem
+ * @param {Object} modulesData - the data set that maps old imports to the new
+ * ones. It is an object like this one:
+ * {
+ *   'ember-array': {
+ *     default: ['@ember/array'],
+ *   },
+ *   'ember-array/mutable': {
+ *     default: ['@ember/array/mutable'],
+ *   },
+ *   'ember-array/utils': {
+ *     A: ['@ember/array', 'A'],
+ *     isEmberArray: ['@ember/array', 'isArray'],
+ *     wrap: ['@ember/array', 'makeArray'],
+ *   },
+ * }
+ * @return {function(fixer: Object): function} a function that applies a fix to
+ * update the import
+ */
+function buildFix(node, modulesData) {
```

Comment:
This function is too long and complicated and should have simple helper functions extracted from it.

Reply:
This function is [existing on master](https://github.com/ember-cli/eslint-plugin-ember/blob/v6.7.1/lib/rules/no-old-shims.js#L273), as a part of a longer function. I've extracted to an util to be reused between two rules. With that explanation, do you think I should implement your suggestion?